metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/NodeMetricsClient.java

```diff
@@ -41,7 +42,7 @@
     private final HttpClient httpClient;
     private final Clock clock;
 
-    private final Map<ConsumerId, Snapshot> snapshots = new HashMap<>();
+    private final Map<ConsumerId, Snapshot> snapshots = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Use `ConcurrentHashMap` instead. I am pretty sure that `Collections.synchronizedMap` may thrown `ConcurrentModificationException` if the map is iterated on concurrently with mutating operations.

Reply:
Yes, you are right.